tecken/api/views.py

```diff
@@ -590,11 +584,25 @@ def upload_files(request):
     if include_bucket_names:
         qs = qs.filter(bucket_name__in=include_bucket_names)
 
-    files = []
-    batch_size = settings.API_FILES_BATCH_SIZE
-    start = (page - 1) * batch_size
-    end = start + batch_size
+    selector = form.cleaned_data["selector"]
+
+    context = {}
+    if "files" in selector:
+        pagination_form = PaginationForm(request.GET)
+        if not pagination_form.is_valid():
+            return http.JsonResponse({"errors": pagination_form.errors}, status=400)
+        page = pagination_form.cleaned_data["page"]
+        context = _upload_files_content(page, qs)
+
+    if "aggregates" in selector:
+        aggregates = _upload_files_aggregates(qs)
+        context["aggregates"] = aggregates
+        context["total"] = aggregates["files"]["count"]
```

Comment:
I'm not wild about using the same API endpoint in two selector modes because it doesn't move us a step towards a better and more usable API. It's fine for now, but if we decide that this solution path helps significantly and we're planning to stick with this until we do the rewrite, we might want to revisit this and break this into two separate API endpoints.

In the meeting, we talked about this a bit and how it's not clear the aggregates are helpful in the first place. Maybe we should ditch them altogether?

Reply:
I had different endpoints before, then turned it into this thinking it'd cleaner. But you're right, the current design is limiting. I'll split them up again

Reply:
About ditching the aggregates, less code is better, so if you say it's not so useful I'd be in favor of ditching it, yes. But would you mind doing that in a future step? 
Btw, do we also want to get rid of that total count on the top-left corner of the table?
<img width="801" alt="Screen Shot 2022-04-01 at 12 22 50 PM" src="https://user-images.githubusercontent.com/5804462/161303861-beae56dc-592e-42ed-9dca-3dd325689e37.png">

Reply:
I don't mind pushing removing the aggregates off to a future step.

Knowing the total count is generally helpful. I think I've used that when filtering.